codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeadersFactory.java

```diff
@@ -0,0 +1,327 @@
+            }
+            int index = HttpHeaderValidationUtil.validateToken(name);
+            if (index != -1) {
+                throw new IllegalArgumentException("a header name can only contain \"token\" characters, " +
+                        "but found invalid character 0x" + Integer.toHexString(name.charAt(index)) +
+                        " at index " + index + " of header '" + name + "'.");
+            }
+        }
+    };
+    private static final ValueValidator<CharSequence> DEFAULT_VALUE_VALIDATOR = new ValueValidator<CharSequence>() {
+        @Override
+        public void validate(CharSequence value) {
+            int index = HttpHeaderValidationUtil.validateValidHeaderValue(value);
+            if (index != -1) {
+                throw new IllegalArgumentException("a header value contains prohibited character 0x" +
+                        Integer.toHexString(value.charAt(index)) + " at index " + index + '.');
+            }
+        }
+    };
+    private static final NameValidator<CharSequence> DEFAULT_TRAILER_NAME_VALIDATOR =
+            new NameValidator<CharSequence>() {
+                @Override
+                public void validateName(CharSequence name) {
+                    DEFAULT_NAME_VALIDATOR.validateName(name);
+                    if (HttpHeaderNames.CONTENT_LENGTH.contentEqualsIgnoreCase(name)
+                            || HttpHeaderNames.TRANSFER_ENCODING.contentEqualsIgnoreCase(name)
+                            || HttpHeaderNames.TRAILER.contentEqualsIgnoreCase(name)) {
+                        throw new IllegalArgumentException("prohibited trailing header: " + name);
+                    }
+                }
+            };
+
+    @SuppressWarnings("unchecked")
+    private static final NameValidator<CharSequence> NO_NAME_VALIDATOR = NameValidator.NOT_NULL;
+    @SuppressWarnings("unchecked")
+    private static final ValueValidator<CharSequence> NO_VALUE_VALIDATOR =
+            (ValueValidator<CharSequence>) ValueValidator.NO_VALIDATION;
+
+    private static final DefaultHttpHeadersFactory DEFAULT = new DefaultHttpHeadersFactory();
+    private static final DefaultHttpHeadersFactory DEFAULT_TRAILER =
+            new DefaultHttpHeadersFactory(DEFAULT_TRAILER_NAME_VALIDATOR, DEFAULT_VALUE_VALIDATOR, false);
+    private static final DefaultHttpHeadersFactory DEFAULT_COMBINING =
+            new DefaultHttpHeadersFactory(DEFAULT.nameValidator, DEFAULT.valueValidator, true);
+    private static final DefaultHttpHeadersFactory DEFAULT_NO_VALIDATION =
+            new DefaultHttpHeadersFactory(NO_NAME_VALIDATOR, NO_VALUE_VALIDATOR, false);
+
+    private final NameValidator<CharSequence> nameValidator;
+    private final ValueValidator<CharSequence> valueValidator;
+    private final boolean combiningHeaders;
+
+    /**
+     * Create a header builder with the default settings.
+     */
+    private DefaultHttpHeadersFactory() {
+        this(DEFAULT_NAME_VALIDATOR, DEFAULT_VALUE_VALIDATOR, false);
```

Comment:
This ctor is used only in one place (line 73). Consider removing it and using all arguments in line 73 to make it easier to spot a difference between `DEFAULT` and `DEFAULT_TRAILER`.